### PR TITLE
Warnint util 4516 v12.1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4516

Describe changes:
- Fix integer warnings `-Wimplicit-int-conversion` for detect*
- Adds ci check for this warning

Final part of #7006 after #7351

Then remains only detect files to fix

Should this be split in multiple commits ?

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on report_failure.

ERROR: QA failed on tlpr1_asan_suri.
  

Pipeline 7671

File: src/detect-content.c [hunk 1]
```diff
@@ -506,36 +506,62 @@ void DetectContentPropagateLimits(Signature *s)
...
                         if (depth && depth > offset_plus_pat) {
-                            uint16_t dist = 0;
+                            int32_t dist = 0;
                             if (cd->flags & DETECT_CONTENT_DISTANCE && cd->distance > 0) {
                                 dist = cd->distance;
                                 SCLogDebug("distance to add: %u. depth + dist %u", dist, depth + dist);
                             }
                             SCLogDebug("depth %u + cd->within %u", depth, cd->within);
-                            depth = cd->depth = depth + cd->within + dist;
+                            if (depth + cd->within + dist < 0 ||
```
Comment (victorjulien) [inline src/detect-content.c:523]: I would like these to go into their own commit

File: src/detect-app-layer-event.c [hunk 2]
```diff
@@ -231,7 +231,11 @@ static int DetectAppLayerEventParseAppP2(DetectAppLayerEventData *data,
             return -3;
         }
     }
-    data->event_id = event_id;
+    if (event_id > UINT8_MAX) {
+        SCLogWarning(SC_ERR_INVALID_SIGNATURE, "app-layer-event keyword's id has invalid value");
```
Comment (victorjulien) [inline src/detect-app-layer-event.c:235]: new error/warning condition, would like a separate commit here

**Comment by catenacyber** [comment]
QA triggers debug assertion at  detect-engine-content-inspection.c:531

(oops)

File: src/detect.c [hunk 3]
```diff
@@ -1150,7 +1149,8 @@ static bool DetectRunTxInspectRule(ThreadVars *tv,
                 TRACE_SID_TXS(s->id, tx, "engine %p match %d", engine, match);
                 if (engine->stream) {
                     can->stream_stored = true;
-                    can->stream_result = match;
+                    // TODO change Callback prototype ?
```
Comment (victorjulien) [inline src/detect.c:1152]: yes I think that makes sense, also in a separate commit

File: src/detect.h [hunk 4]
```diff
@@ -1021,7 +1021,7 @@ typedef struct HttpReassembledBody_ {
 typedef struct SignatureNonPrefilterStore_ {
     SigIntId id;
     SignatureMask mask;
-    uint8_t alproto;
+    AppProto alproto;
```
Comment (victorjulien) [inline src/detect.h:1024]: does this change the size of the struct? guess not if `AppProto` is uint16_t, but if it is an enum it would

**Review by victorjulien — CHANGES_REQUESTED** [review]
Requested separate commits in a few places, the rest can be in a single commit

File: src/detect-content.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-content.c:523]: ok, looks relevant

File: src/detect-app-layer-event.c (hunk 2 above)
Comment (catenacyber) [reply src/detect-app-layer-event.c:235]: ok

File: src/detect.c (hunk 3 above)
Comment (catenacyber) [reply src/detect.c:1152]: that makes a big commit...

Comment (catenacyber) [reply src/detect.c:1152]: (done in next version)

File: src/detect.h (hunk 4 above)
Comment (catenacyber) [reply src/detect.h:1024]: `typedef uint16_t AppProto;` ;-)
